contrib/subtree/Makefile

```diff
@@ -94,7 +94,7 @@ $(GIT_SUBTREE_TEST): $(GIT_SUBTREE)
 	cp $< $@
 
 test: $(GIT_SUBTREE_TEST)
-	$(MAKE) -C t/ test
+	$(MAKE) -C t/ all
 
```

Comment:
Is this just a bug in the existing makefile? Did `make -C contrib/subtree test` just not work before?

This might be a good one-liner to send upstream, even if we don't choose to do the rest of the change. I would split them into two commits, first this change, then the changes to `ci/`.

Reply:
I _think_ the idea here is that `all` defaults to running `DEFAULT_TEST_TARGET`, which is set to `prove` in the CI runs. And `test` is a _different_ target.

An alternative I briefly considered was to call `make` twice: once to "build" `git-subtree`, and then the test run. But that would be rather ugly because the tests require the `git-subtree` file to be copied to `../..`, which the `test` target does, thanks to the prerequisite `$(GIT_SUBTREE_TEST)`.

In short: I concur with @derrickstolee that it would be best to split out the change to the `Makefile`, and the commit message should probably mention that we're aligning with how Git's top-level `Makefile` defines its own `test` target (it also calls `$(MAKE) -C t/ all`, and then also mention that we want to be able to call this `make` target in `contrib/subtree` in the CI runs, where `DEFAULT_TEST_TARGET` is defined as `prove`.

Reply:
> it would be best to split out the change to the `Makefile`

Sounds good! Depending on whether you'd like me to make the change or not, should changing the use of `pre-clean` be part of the same commit, or a separate one?

Reply:
It can definitely be not only a separate commit, but an independent patch series. And nothing about that patch series would be urgent: since we lost Azure Pipelines' powerful analysis/statistics/UI for tests (see e.g. [here](https://dev.azure.com/git-for-windows/git/_build/results?buildId=59006&view=ms.vss-test-web.build-test-results-tab)), we do not _actually_ care about the full test results to be maintained. We are only interested in them if anything fails. And since we only run the `subtree` tests if Git's own test suite passed, that desire is still addressed.

However, should we ever decide to _always_ run the `subtree` tests (even if Git's test suite failed, in which we obviously would still want to fail the test, i.e. record the exit status indicating failure in some way), we would definitely need to get rid of the automatic `pre-clean` dependency.

For now, though, there is no point in holding this PR up, I think. It would be good to have it in `-rc2` (I would probably reorder the `pick`s and the `merge` somewhat, so that the `Merge 'readme'` commit still comes on top).